src/descriptor/key.rs

```diff
@@ -314,6 +314,44 @@ impl DescriptorSecretKey {
 
         Ok(pk)
     }
+
+    /// Whether or not this key has multiple derivation paths.
+    pub fn is_multipath(&self) -> bool {
+        match *self {
+            DescriptorSecretKey::Single(..) | DescriptorSecretKey::XPrv(..) => false,
+            DescriptorSecretKey::MultiXPrv(_) => true,
+        }
+    }
```

Comment:
This seems redundant with `MiniscriptKey`'s `num_der_paths` introduced later on, but i kept it here in order to also have it for private keys.